src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/ProductUniqueDataRepository.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Pim\Bundle\CatalogBundle\Doctrine\ORM\Repository;
+
+use Doctrine\ORM\EntityRepository;
+use Pim\Component\Catalog\Model\ProductInterface;
+use Pim\Component\Catalog\Model\ProductValueInterface;
+use Pim\Component\Catalog\Repository\ProductUniqueDataRepositoryInterface;
+
+/**
+ * Product unique data repository. Please see {@see Pim\Component\Catalog\Model\ProductUniqueDataInterface}
+ * for more information.
+ *
+ * @author    Julien Janvier <julien.janvier@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class ProductUniqueDataRepository extends EntityRepository implements ProductUniqueDataRepositoryInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function uniqueDataNotExistsInThisProduct(ProductValueInterface $value, ProductInterface $product)
+    {
+        $queryBuilder = $this->createQueryBuilder('ud')
+            ->select('COUNT(ud)')
+            ->where('ud.attribute = :attribute')
+            ->andWhere('ud.rawData = :data')
+        ;
+
+        $parameters = [
+            'attribute' => $value->getAttribute(),
+            'data' => $value->__toString(),
+        ];
+
+        if (null !== $product->getId()) {
```

Comment:
This seems pretty fragile, if the product has no id yet (new product not persisted yet), we check if the data is not used on all the products of the DB (not what the method naming let me expect).

I would split in two methods, for instance,
 - uniqueDataNotExists
 - uniqueDataNotExistsInThisProduct

At the end i don't get why we check for this product and not for all products